chrome/content/zotero/xpcom/style.js

```diff
@@ -399,6 +431,72 @@ Zotero.Styles = new function() {
 			}
 		});
 	}
+	
+	/**
+	 * Populate menulist with locales
+	 */
```

Comment:
> Is that right?

Yes.

> What exactly is the intention with the "preferred" locale here? From what I can tell, that's the most-recently-used locale, correct?

Well, as discussed in the pull request, we decided to go with two preferences. I also not always use the locale from the pref (e.g. for the site-specific settings window, see https://github.com/zotero/zotero/pull/623/files#diff-dea0c6d13540d741eafa89452186235aR89 ).

> And no, I don't think we should (re)move the preferred/fallback language from the general list.

These extra fallback and preferred locale entries are only added if they are not already in the general list.